src/contract/Signature.ts

```diff
@@ -15,12 +19,15 @@ Different types which can be used to sign transaction or data item
 - Signer - arbundles specific class which allows to sign data items (only this type can be used when bundling is enabled and data items 
   are being created)
 */
-export type SignatureProvider = ArWallet | CustomSignature | Signer;
+export type SignatureProvider = ArWallet | CustomSignature | BundlerSigner;
 
 export class Signature {
   signer: SigningFunction;
-  type: SignatureType;
+  readonly type: SignatureType;
   readonly warp: Warp;
+  private readonly signatureProviderType: 'CustomSignature' | 'ArWallet' | 'BundlerSigner';
+  private readonly wallet;
+  private cachedAddress?: string;
 
   constructor(warp: Warp, walletOrSignature: ArWallet | CustomSignature) {
```

Comment:
shouldn't we add BundlerSigner type here?

Reply:
Yep, good catch